### PR TITLE
Fix validations based on fields imported from ECS

### PR DESCRIPTION
**Pull request by jsoriano**
We inject fields from ECS on different scenarios: when generating documentation, when building packages, or when validating documents in tests.

On #1335 we did a refactor around this, to fix some validation issues, in a way that more code is reused between all these uses. After this change, validation used field definitions that include the already resolved external fields, but we weren't including there information that was used by validators, what included:
* Validation of expected values.
* Validation of allowed values.

So these validations haven't been executed since then.

Tests have been included to try to avoid regressions related to this in the future.

Fix https://github.com/elastic/elastic-package/issues/1439

### CODE REVIEW

File: internal/fields/dependency_manager.go [hunk 1]
```diff
@@ -318,17 +328,28 @@ func transformImportedField(fd FieldDefinition) common.MapStr {
...
+		}
+
+		if len(fd.AllowedValues) > 0 {
+			m["allowed_values"] = fd.AllowedValues
+		}
+
+		if len(fd.ExpectedValues) > 0 {
+			m["expected_values"] = fd.ExpectedValues
+		}
+	}
```
Comment (mrodm) [inline internal/fields/dependency_manager.go:352]: This means that these fields (normalize, allowed_values, expected_values) won't be added to each field definition when the package are built, is that right?

File: internal/fields/validate.go [hunk 2]
```diff
@@ -163,6 +163,8 @@ func CreateValidatorForDirectory(fieldsParentDir string, opts ...ValidatorOption
 
 func createValidatorForDirectoryAndPackageRoot(fieldsParentDir string, finder packageRootFinder, opts ...ValidatorOption) (v *Validator, err error) {
 	v = new(Validator)
+	// In validator, inject fields with settings used for validation, such as `allowed_values`.
+	v.injectFieldsOptions.IncludeValidationSettings = true
```
Comment (mrodm) [inline internal/fields/validate.go:167]: If it is set to true here, every run of validation would use those fields, even in built packages ?

**Comment by jsoriano** [comment]
test integrations

File: internal/fields/validate.go (hunk 2 above)
Comment (mrodm) [reply internal/fields/validate.go:167]: Talked offline, and this should be just used for validators. Built packages do not use this one.

File: internal/fields/dependency_manager.go (hunk 1 above)
Comment (jsoriano) [reply internal/fields/dependency_manager.go:352]: Yes, this is not used in Fleet or Elasticsearch.

**Review by mrodm — APPROVED** [review]
Thanks!!

**Pull request merged**